### DISCUSSION TITLE
Why do I get 'Query requires iterating over all vertices' when checking for a connection?

### DISCUSSION THREAD

**Discussion by berniedud** [JanusGraph/janusgraph#4742] [Q&A] [answered]
This is a copy of [this StackOverflow question](https://stackoverflow.com/questions/79271356/why-do-i-get-query-requires-iterating-over-all-vertices-warning-when-checking).

We are using JanusGraph (1.0.0) with a python fastAPI-based API, on kubernetes, to maintain and query our gremlin graph. Using gremlinpython, we periodically send this query for the sole purpose of checking the connection between API and DB server:

`result = graph.inject(1).next()`, where graph is our graph traversal object.

Our understanding is that this query is equivalent to sending something like SELECT 1; to a Postgres (other RDBMS exist) server. It creates a traversal with a single value, then collects that value.

When we issue this query, we get a warning from the JanusGraph server, like this:

`10:18:49 WARN  org.janusgraph.graphdb.transaction.StandardJanusGraphTx$3.execute - Query requires iterating over all vertices [[]]. For better performance, use indexes`

Is our understanding wrong? Does this query actually iterate over our graph (which is not even referenced in this query)? Or is this warning a bug?

The problem we have with this warning is that we may well be committing this crime (not using indexes) elsewhere in the code, but there are so many spurious warnings that we will never be able to recognise a real one.

**Comment by porunov** [comment, marked as answer]
Perhaps there was a bug in 1.0.0, but the query you showed shouldn't trigger this warning at least for the version `1.1.0`. If it triggers this warning, please, open an issue so that we can take a closer look. However, for now I confirmed that `g.inject(1).next()` is not issuing full-scan warning in `master` branch (unreleased `1.2.0`).
In case you can't update to the latest JanusGraph version for any reason, you could use `g.tx().isOpen()` check that should simply return `true`.
In case you want to ensure you never trigger full-scan queries you can set configuration `query.force-index` to `true` and it will issue an exception for any full-scan query.
